girder-dandi-archive/girder_dandi_archive/util.py

```diff
@@ -23,3 +25,40 @@ def validate_dandiset_identifier(dandiset_identifier):
 
 def get_or_create_drafts_collection():
     return Collection().createCollection(DANDI_DRAFTS_COLLECTION_NAME, reuseExisting=True)
+
+
+def find_dandiset_by_identifier(identifier):
+    """Use a unique identifier to find a dandiset in the dandiset drafts collection."""
+    drafts = get_or_create_drafts_collection()
+    return Folder().findOne({"parentId": drafts["_id"], "meta.dandiset.identifier": identifier})
+
+
+def find_dandiset(query):
```

Comment:
why is this added? doesn't seem like it is used. also, what form should `query` take?

Reply:
This is a general helper function which may be useful to add into several places in `rest.py`, but I haven't done that yet as I wanted consensus  from people. I encountered a weird bug from not including the `{"parentId": drafts["_id"], ...}` portion of a mongo query. This would just abstract that out, and `query` would be any valid mongo query.